src/core/symbology/qgsstyle.cpp

```diff
@@ -1219,17 +1208,18 @@ QgsSymbolGroupMap QgsStyle::smartgroupsListMap()
   char *query = sqlite3_mprintf( "SELECT * FROM smartgroup" );
 
   // Now run the query and retrieve the group names
-  sqlite3_stmt *ppStmt = nullptr;
-  int nError = sqlite3_prepare_v2( mCurrentDB, query, -1, &ppStmt, nullptr );
+  sqlite3_statement_unique_ptr statement;
+  int nError;
+  mCurrentDB.prepare( query, nError );
 
   QgsSymbolGroupMap groupNames;
-  while ( nError == SQLITE_OK && sqlite3_step( ppStmt ) == SQLITE_ROW )
+  while ( nError == SQLITE_OK && sqlite3_step( statement.get() ) == SQLITE_ROW )
   {
-    QString group = QString::fromUtf8( reinterpret_cast< const char * >( sqlite3_column_text( ppStmt, SmartgroupName ) ) );
-    groupNames.insert( sqlite3_column_int( ppStmt, SmartgroupId ), group );
+    QString group = QString::fromUtf8( reinterpret_cast< const char * >( sqlite3_column_text( statement.get(), SmartgroupName ) ) );
+    groupNames.insert( sqlite3_column_int( statement.get(), SmartgroupId ), group );
   }
 
-  sqlite3_finalize( ppStmt );
+  statement.release();
```

Comment:
Not really necessary, but no harm either

Reply:
Actually, yes it does harm. Just removed all those, they'll prevent cleanup from happening (releasing just sets the internal pointer to `nullptr` without calling the deletor).

Reply:
Oh man I wasted a couple of hours when I made that same mistake recently too - using .release() instead of .reset(). It's really tricky to pick up while reading over code.

IMO clang/gcc should throw a warning when .release() is called and nothing captures the returned value. There's no valid use case I can see for this situation.